flow/texture_image.cc

```diff
@@ -31,11 +32,60 @@ inline GLint ToGLDataFormat(TextureImageDataFormat dataFormat) {
   return GL_NONE;
 }
 
+inline GrPixelConfig ToGrPixelConfig(TextureImageDataFormat dataFormat) {
+  switch (dataFormat) {
+    case TextureImageDataFormat::UnsignedByte:
+      return kRGBA_8888_GrPixelConfig;
+    case TextureImageDataFormat::UnsignedShort565:
+      return kRGB_565_GrPixelConfig;
+  }
+  return kUnknown_GrPixelConfig;
+}
+
+sk_sp<SkImage> TextureImageCreate(GrContext* context, const SkBitmap& bitmap) {
+  if (context == nullptr) {
+    return nullptr;
+  }
+
+  if (bitmap.drawsNothing()) {
+    return nullptr;
+  }
+
+  TextureImageDataFormat dataFormat = TextureImageDataFormat::UnsignedByte;
+  TextureImageFormat imageFormat = TextureImageFormat::RGBA;
+
+  switch (bitmap.colorType()) {
+    case kRGB_565_SkColorType:
+      dataFormat = TextureImageDataFormat::UnsignedShort565;
+      imageFormat = TextureImageFormat::RGB;
+      break;
+    case kRGBA_8888_SkColorType:
+      dataFormat = TextureImageDataFormat::UnsignedByte;
+      imageFormat = TextureImageFormat::RGBA;
+      break;
+    default:
+      // Add more as supported.
```

Comment:
The fallback paths are also necessary for the test targets that don't setup a GL context on the resource loading thread.